models/pooltogether/ethereum/pooltogether_v4_ethereum_prize_structure.sql

```diff
@@ -0,0 +1,109 @@
+{{ config(
+    alias ='prize_structure',
+    materialized = 'incremental',
+    file_format = 'delta',
+    incremental_strategy = 'merge',
+    unique_key = ['tx_hash', 'network'],
+    post_hook='{{ expose_spells(\'["ethereum"]\',
+                                "project",
+                                "pooltogether_v4",
+                                \'["bronder"]\') }}'
+)}}
+
+WITH
+  --Calculate proze structure for Ethereum network per drawID
+prizeDistribution AS (
+    --ETHEREUM POST DPR
+    SELECT call_tx_hash                                                        AS tx_hash,
+           call_block_time                                                     AS block_time,
+           'Ethereum'                                                          AS network,
+           drawId,
+           get_json_object(output_0, '$.bitRangeSize')                         AS bitRange,
+           substr(
+                   get_json_object(output_0, '$.tiers'),
+                   2,
+                   length(get_json_object(output_0, '$.tiers')) - 2
+               )                                                               AS tiers,
+           CAST(get_json_object(output_0, '$.dpr') AS int) / power(10, 6)      AS dpr,
+           CAST(get_json_object(output_0, '$.prize') AS double) / power(10, 6) AS prize
+    FROM
+      {{ source('pooltogether_v4_ethereum', 'PrizeTierHistoryV2_call_getPrizeTier')}}
+    WHERE call_success = true
+    {% if is_incremental() %}
+      AND call_block_time >= date_trunc("day", now() - interval '1 week')
+    {% endif %}
+
+    UNION ALL
+
+    --ETHEREUM PRE DPR
+    SELECT evt_tx_hash                                                                  AS tx_hash,
+           evt_block_time,
+           'Ethereum'                                                                   AS network,
+           drawId,
+           get_json_object(prizeDistribution, '$.bitRangeSize')                         AS bitRange,
+           substr(
+                   get_json_object(prizeDistribution, '$.tiers'),
+                   2,
+                   length(get_json_object(prizeDistribution, '$.tiers')) - 2
+               )                                                                        AS tiers,
+           0                                                                            AS dpr,
+           CAST(get_json_object(prizeDistribution, '$.prize') AS double) / power(10, 6) AS prize
+    FROM
+      {{ source('pooltogether_v4_ethereum', 'PrizeDistributionBuffer_evt_PrizeDistributionSet')}}
+    WHERE drawID < 447 --DPR On Ethereum started on draw 447
+    {% if is_incremental() %}
+      AND evt_block_time >= date_trunc("day", now() - interval '1 week')
+    {% endif %}
+),
+
+detailedPrizeDistribution AS (
+    SELECT tx_hash,
+           block_time,
+           network,
+           drawId,
+           CAST(bitRange AS int)                   AS bitRange,
```

Comment:
Maybe for consistency:
```suggestion
           drawId                                  AS draw_id,
           CAST(bitRange AS int)                   AS bit_range,
```
also you can rename the CTE in this way.

Reply:
Agreed, should be fixed now